contracts/minitoken/solidity/contracts/app/MiniToken.sol

```diff
@@ -0,0 +1,271 @@
+pragma solidity ^0.8.9;
+
+import "@hyperledger-labs/yui-ibc-solidity/contracts/core/IBCHandler.sol";
+import "@hyperledger-labs/yui-ibc-solidity/contracts/core/IBCHost.sol";
+import "@hyperledger-labs/yui-ibc-solidity/contracts/core/IBCModule.sol";
+import "@hyperledger-labs/yui-ibc-solidity/contracts/core/types/Channel.sol";
+import "@hyperledger-labs/yui-ibc-solidity/contracts/lib/Bytes.sol";
+import "../lib/Packet.sol";
+
+contract MiniToken is IModuleCallbacks {
+    IBCHandler ibcHandler;
+    IBCHost ibcHost;
+
+    using Bytes for *;
+
+    address private owner;
+
+    constructor(IBCHost host_, IBCHandler ibcHandler_) public {
+        owner = msg.sender;
+
+        ibcHost = host_;
+        ibcHandler = ibcHandler_;
+    }
+
+    event Mint(address indexed to, uint256 amount);
+
+    event Burn(address indexed from, uint256 amount);
+
+    event Transfer(address indexed from, address indexed to, uint256 amount);
+
+    event SendTransfer(
+        address indexed from,
+        address indexed to,
+        string sourcePort,
+        string sourceChannel,
+        uint64 timeoutHeight,
+        uint256 amount
+    );
+
+    modifier onlyOwner() {
+        require(msg.sender == owner, "caller is not the owner");
+        _;
+    }
+
+    modifier onlyIBC() {
+        require(
+            msg.sender == address(ibcHandler),
+            "caller is not the ibcHandler"
+        );
+        _;
+    }
+
+    function sendTransfer(
+        uint64 amount,
+        address receiver,
+        string calldata sourcePort,
+        string calldata sourceChannel,
+        uint64 timeoutHeight
+    ) external {
+        require(_burn(msg.sender, amount));
+
+        _sendPacket(
+            MiniTokenPacketData.Data({
+                amount: amount,
+                sender: abi.encodePacked(msg.sender),
+                receiver: abi.encodePacked(receiver)
+            }),
+            sourcePort,
+            sourceChannel,
+            timeoutHeight
+        );
+        emit SendTransfer(
+            msg.sender,
+            receiver,
+            sourcePort,
+            sourceChannel,
+            timeoutHeight,
+            amount
+        );
+    }
+
+    mapping(address => uint256) private _balances;
+
+    function mint(address account, uint256 amount) external onlyOwner {
+        require(_mint(account, amount), "invalid address");
+    }
+
+    function burn(uint256 amount) external {
+        require(_burn(msg.sender, amount), "invalid address");
+    }
+
+    function transfer(address to, uint256 amount) external {
+        bool res;
+        string memory message;
+        (res, message) = _transfer(msg.sender, to, amount);
+        require(res, message);
+    }
+
+    function balanceOf(address account) public view returns (uint256) {
+        require(account != address(0), "Token: invalid address");
```

Comment:
I know it's a small detail, but why is the case `account == 0` handled specially?
Users can still mint or transfer a token to any non-zero invalid addresses, can't they?

Reply:
@siburu 
You're absolutely right. There's no need for special handling here. Thank you.

Reply:
@3100 Thank you and I'm sorry that I have to say that `_mint`, `_burn` and `_transfer` are also checking if `account == 0`. Wouldn't it be better to remove these as well?

Reply:
@siburu 
Sorry for the delay in responding. Considering that we can apply the MiniToken design to other ledgers, it was strange to consider only the zero address invalid.
Fixed it.